_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/WordpressApi.php

```diff
@@ -101,30 +113,42 @@ public function getContentByCategory($categories, $max_results, $select_logic) {
       }
     }
 
-    try {
-      // Retrieve the WP posts from the $feed_url and decode the JSON into a
-      // $results array.
-      $request = $this->client->request('GET', $feed_url, ['query' => $query]);
+    $imploded_categories = implode('', $categories);
+    $cid = "wordpress_api:by_category:per_page_{$query['per_page']}-logic_{$query['logic']}-categories_{$imploded_categories}";
 
-      $response = $request->getBody()->getContents();
+    // Serve the cached Wordpress API data if available.
+    if ($cache = \Drupal::cache()->get($cid)) {
+      $data = $cache->data;
+    }
+    // If this is uncached or the cache is invalid, fetch fresh data.
+    else {
+      try {
+        // Retrieve the WP posts from the $feed_url and decode the JSON into a
+        // $results array.
+        $request = $this->client->request('GET', $feed_url, ['query' => $query]);
 
-      $results = json_decode($response);
+        $response = $request->getBody()->getContents();
 
-      // Pass the WP posts returned in $results, and get the processed/formatted
-      // results from getContentCompositeMultiple().
-      $items = $this->getContentCompositeMultiple($results);
+        $results = json_decode($response);
 
-      return $items;
-    }
-    catch (\Exception $e) {
-      \Drupal::logger('rhd_assemblies')->error(
-        "Exception while calling Wordpress API: @message", [
-          '@message' => $e->getMessage(),
-        ]
-      );
-
-      return [];
+        // Pass the WP posts returned in $results, and get the processed/formatted
+        // results from getContentCompositeMultiple().
+        $data = $this->getContentCompositeMultiple($results);
+        // Persist this data to the wordpress_api cache bin for an hour.
+        \Drupal::cache()->set($cid, $data, time() + 3600);
```

Comment:
@jordanpagewhite Same question here as for the Download Manager integration. Can we extract the TTL to a configuration value with some sane default applied if it's not set?